File: server/python/app/infrastructure/mappers/payment_details/protection_eligibility_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.protection_eligibility import ProtectionEligibility
from onlinepayments.sdk.domain.protection_eligibility import ProtectionEligibility as ProtectionEligibilitySdk

class ProtectionEligibilityMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[ProtectionEligibilitySdk]) -> Optional[ProtectionEligibility]:
        if response is None:
            return None

        dto = ProtectionEligibility()
        dto.eligibility = response.eligibility
        dto.type = response.type

        return dto

File: server/ruby/app/infrastructure/mappers/payment_details/protection_eligibility_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class ProtectionEligibilityMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::ProtectionEligibility.new(
            eligibility: response.eligibility,
            type: response.type
          )
        end
      end
    end
  end
end